src/visualizers/panels/ExecutionIndex/ExecutionIndexControl.js

```diff
@@ -39,8 +41,26 @@ define([
         this._logger.debug('ctor finished');
     };
 
+    ExecutionIndexControl.prototype = Object.create(Execute.prototype);
+
+    ExecutionIndexControl.prototype._deleteExecution = function (id) {
+        let node = this._client.getNode(id),
+            name = '';
+        if (node) {
+            name = node.getAttribute('name');
+        }
+
+        this._client.startTransaction(`Deleted ${name} (${id}) execution.`);
+        if (this.isRunning(node)) {
+            this.stopExecution(id);
+        }
+        this._client.deleteNode(id);
+        this._client.completeTransaction();
+    };
+
     ExecutionIndexControl.prototype._initWidgetEventHandlers = function () {
         this._widget.setDisplayedExecutions = this.setDisplayedExecutions.bind(this);
+        this._widget._deleteExecution = this._deleteExecution.bind(this);
```

Comment:
Minor naming convention issue - this should probably be `this._widget.deleteExecution = this._deleteExecution.bind(this);` to match the naming convention used for methods added to the widget from the controller. 

Technically, a case could be made that they are kinda private methods and therefore could/should have a `_` prefix in the name. I say "kinda private" because they are only called within the widget but they are implemented on the controller. That is, they are a bit odd since they are really defining the API defined by the controller for editing the model. Anyway, it would be good to be consistent with the convention currently used.

Reply:
The method binds have been changed to 
`
this._widget.deleteExecution = this._deleteExecution.bind(this);
`